### PR TITLE
Add end to end test embed inside block with locking all.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Adds an end 2 end test for the issue fixed in https://github.com/WordPress/gutenberg/pull/15866.

## How has this been tested?
I checked the end 2 end tests pass.
I reverted https://github.com/WordPress/gutenberg/pull/15866 and I verified the end to end tests fail.

### CODE REVIEW

File: packages/e2e-tests/specs/embedding.test.js [hunk 1]
```diff
@@ -229,6 +231,21 @@ describe( 'Embedding content', () => {
...
 	} );
 
+	it( 'embed blocks should work inside blocks with locking all', async () => {
+		await activatePlugin( 'gutenberg-test-innerblocks-locking-all-embed' );
+		await createNewPost();
+		await insertBlock( 'Test Inner Blocks Locking All Embed' );
+		const embedInputSelector = '.components-placeholder__input[aria-label="Embed URL"]';
+		await page.waitForSelector( embedInputSelector );
+		await page.click( embedInputSelector );
+		// This URL can't be embedded, but without the trailing slash, it can.
```
Comment (gziolo) [inline packages/e2e-tests/specs/embedding.test.js]: Should URL have the trailing slash? This comment is confusing.

File: packages/e2e-tests/specs/embedding.test.js [hunk 2]
```diff
@@ -229,6 +231,21 @@ describe( 'Embedding content', () => {
...
+		await page.keyboard.type( 'https://twitter.com/notnownikki' );
+		await page.keyboard.press( 'Enter' );
+		// The twitter block should appear correctly.
+		await page.waitForSelector( 'figure.wp-block-embed' );
+		await deactivatePlugin( 'gutenberg-test-innerblocks-locking-all-embed' );
```
Comment (gziolo) [inline packages/e2e-tests/specs/embedding.test.js]: I think this test should be wrapped with its own `describe` and the plugin should be activated in `before` and deactivated in `after` methods to ensure they are properly handled when the test fails.

**Review by gziolo — COMMENTED** [review]
I hoped we can avoid creating another plugin to test this scenario, but it's fine as is, as well. There is one thing to resolve though, I think we need to move this new test to its own file inside `plugins` directory. There are external projects I know about which can't install plugins so they need to whitelist all tests that use plugins. It's much easier when they can do it based on the fact that all of them are grouped in one folder.

File: packages/e2e-tests/specs/embedding.test.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/e2e-tests/specs/embedding.test.js]: We need the trailing slash the comment was clarified 👍

**Comment by jorgefilipecosta** [comment]
Hi @gziolo, this PR was rebased and your suggestions addressed. Let me know if you think it is ready to land :)

File: packages/e2e-tests/plugins/inner-blocks-locking-all-embed.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Registers a custom script for the plugin.
+ */
+function enqueue_container_without_paragraph_plugin_script() {
+	wp_enqueue_script(
+		'gutenberg-test-inner-blocks-locking-all-embed',
+		plugins_url( 'inner-blocks-locking-all-embed/index.js', __FILE__ ),
+		array(
+			'wp-blocks',
```
Comment (gziolo) [inline packages/e2e-tests/plugins/inner-blocks-locking-all-embed.php]: The list of dependencies still needs to be updated.

**Review by gziolo — APPROVED** [review]
The list of the script dependencies needs to be updated.

Otherwise, everything looks good. Thanks for applying updated 👍

File: packages/e2e-tests/specs/embedding.test.js (hunk 1 above)
Comment (gziolo) [reply packages/e2e-tests/specs/embedding.test.js]: Looking at code I assume, we don't need it
Travis is happy 🤷‍♂

**Pull request merged**